src/datalab_app_plugin_insitu/apps/nmr/blocks.py

```diff
@@ -76,74 +76,108 @@ def get_available_folders(self, file_path: Path) -> List[str]:
+        else:
+            self.data.pop("max_experiments", None)
+            self.data.pop("end_exp", None)
+
         if not all([nmr_folder_name, echem_folder_name]):
             raise ValueError("Both NMR and Echem folder names must be specified")
 
-        start_exp = int(self.data.get("start_exp", self.defaults["start_exp"]))
-        end_exp = self.data.get("end_exp", self.defaults["end_exp"])
-        if end_exp is not None:
-            end_exp = int(end_exp)
-        step_exp = int(self.data.get("step_exp", self.defaults["step_exp"]))
-        exclude_exp = self.data.get("exclude_exp", self.defaults["exclude_exp"])
-
         try:
-            result = process_local_data(
-                folder_name=str(file_path),
-                nmr_folder_name=nmr_folder_name,
-                echem_folder_name=echem_folder_name,
-                start_at=start_exp,
-                end_at=end_exp,
-                step=step_exp,
-                exclude_exp=exclude_exp,
+            nmr_folder_name = self.data.get("nmr_folder_name")
+            echem_folder_name = self.data.get("echem_folder_name")
+
+            if not all([nmr_folder_name, echem_folder_name]):
+                raise ValueError("Both NMR and Echem folder names are required")
+
+            start_exp = int(self.data.get("start_exp", self.defaults["start_exp"]))
+            end_exp = self.data.get("end_exp", self.defaults["end_exp"])
+            if end_exp is not None:
+                end_exp = int(end_exp)
+            step_exp = int(self.data.get("step_exp", self.defaults["step_exp"]))
+            exclude_exp_str = self.data.get("exclude_exp", self.defaults["exclude_exp"])
+            exclude_exp = []
+            if exclude_exp_str and exclude_exp_str.strip():
+                try:
+                    exclude_exp = [int(x.strip()) for x in exclude_exp_str.split(",") if x.strip()]
+                except ValueError:
+                    exclude_exp = []
+
+            try:
+                result = process_local_data(
+                    folder_name=str(file_path),
+                    nmr_folder_name=nmr_folder_name,
+                    echem_folder_name=echem_folder_name,
+                    start_at=start_exp,
+                    end_at=end_exp,
+                    step=step_exp,
+                    exclude_exp=exclude_exp,
+                )
+
+            except FileNotFoundError as e:
+                raise FileNotFoundError(f"Folder not found: {str(e)}")
+
+            except Exception as e:
+                raise RuntimeError(f"Error processing data: {str(e)}")
+
+            nmr_data = result["nmr_spectra"]
+            ppm_values = np.array(nmr_data.get("ppm", []))
+
+            ppm1 = self.data["ppm1"] = min(ppm_values)
+            ppm2 = self.data["ppm2"] = max(ppm_values)
+
+            self.data.update(
+                {
+                    "nmr_data": result["nmr_spectra"],
+                    "echem_data": result.get("echem", {}),
```

Comment:
Likely this is why you and @be-smith were seeing errors around DocumentTooLarge (the limit in MongoDB is ~16 MB per document) --- anything in `self.data` will be saved to the database, and we really do not want the raw spectra saved like this. They are not even being loaded again from the database, so there's really no point.